artifacts/scripts/util.sh

```diff
@@ -787,169 +759,47 @@ function squash() {
-            rm -rf ./vendor/k8s.io
-            # see https://github.com/kubernetes/kubernetes/issues/45693
-            rm -rf ./vendor/github.com/google/gofuzz
-            # go-openapi is shared between apiserver and apimachinery
-            rm -rf ./vendor/github.com/go-openapi
-        fi
-    fi
-
-    git add Godeps/Godeps.json
-    git clean -f Godeps # clean possible new files missing in branch (e.g. README)
-    git add vendor/ --ignore-errors &>/dev/null || true
-
-    # check if there are new contents
-    if git-index-clean; then
-        echo "Godeps.json hasn't changed!"
-    else
-        echo "Committing vendor/ and Godeps/Godeps.json."
-        git commit -q -m "sync: update godeps"
-    fi
-
-    # nothing should be left
-    ensure-clean-working-dir
-}
+    local deps_array=()
+    IFS=',' read -a deps_array <<< "$®{1}"
+    local dep_count=${#deps_array[@]}
+    local base_package=${2}
 
-# update the dependencies to the version checked out
-update-deps-in-gomod() {
-    if [ ! -f Godeps/Godeps.json ]; then
-        return 0
+    # drop peer references
+    if grep -q "// BEGIN STAGING REPLACE" go.mod; then
+    	sed -i '/\/\/ BEGIN STAGING REPLACE/,/\/\/ END STAGING REPLACE/d' your_file
     fi
 
-    local base_package=${2}
-    local is_library=${3}
-    local deps=${1}
-    local commit_msg_tag="${4}"
-    local deps_array=()
-    IFS=',' read -a deps_array <<< "${1}"
-    local dep_count=${#deps_array[@]}
     for (( i=0; i<${dep_count}; i++ )); do
         local dep="${deps_array[i]%%:*}"
-        local dep_commit=$(cd ../${dep}; git rev-parse HEAD)
-        if [ -z "${dep_commit}" ]; then
-            echo "Couldn't find kube commit for current HEAD. Odd."
-            return 1
-        fi
-
-        # get old dependency hash in Godeps/Godeps.json
-        local old_dep_commit=$(jq -r '.Deps[] | select(.ImportPath | startswith("'${base_package}/${dep}'/") or . == "'${base_package}/${dep}'") | .Rev' Godeps/Godeps.json | tail -n 1)
-        if [ -n "${old_dep_commit}" ]; then
-            if [ "${old_dep_commit}" != "${dep_commit}" ]; then
-                # it existed before => replace with the new value
-                echo "Updating k8s.io/${dep} dependency to ${dep_commit}."
-                sed -i "s/${old_dep_commit}/${dep_commit}/g" Godeps/Godeps.json
-            fi
-        elif git grep -w -q "k8s.io/${dep}"; then
-            # revert changes and fall back to full vendoring
-            echo "Found new dependency k8s.io/${dep}. Switching to full vendoring."
-            git checkout -q HEAD Godeps/Godeps.json
-            update_full_gomod "${deps}" "${base_package}" "${is_library}" "${commit_msg_tag}"
-            return $?
-        else
-            echo "Ignoring k8s.io/${dep} dependency because it seems not to be used."
-        fi
+        local dep_commit=$(cd ../${dep}; git show -q --pretty='format:v0.0.0-%cd-%H' --date='format:%Y%m%d%H%M%S')
```

Comment:
using %h and --abbrev=12 gets the format we want:

```
git show -q --pretty='format:v0.0.0-%cd-%h' --date='format:%Y%m%d%H%M%S' --abbrev=12
```

Reply:
fixed